osbenchmark/cloud_provider/vendors/s3_data_producer.py

```diff
@@ -44,20 +45,33 @@ def __init__(self, bucket:str, keys, client_options: dict, data_dir=None) -> Non
         except Exception as e:
             print(f"Error: {e}")
 
-    def _get_keys(self):
-        rsl = list()
-        if self.keys[-1] == '*':
-            keys = self.s3_client.list_objects(Bucket=self.bucket, Prefix=self.keys[:-1])
-            for key in keys['Contents']:
-                rsl.append(key['Key'])
+    def _get_next_key(self):
+        if len(self.keys) > 2 and self.keys.endswith('**'):
+            processed_keys = set()
+            while True:
+                response = self.s3_client.list_objects(Bucket=self.bucket, Prefix=self.keys[:-2])
+                for object in response['Contents']:
+                    key = object['Key']
+                    if key not in processed_keys:
+                        processed_keys.add(key)
+                        size = self.s3_client.head_object(Bucket=self.bucket, Key=key)['ContentLength']
+                        if size == 0:
+                            return
+                        yield key
```

Comment:
should we add a comment here that the streaming terminates here when empty file is encountered?